yaptide/celery/utils/manage_tasks.py

```diff
@@ -96,40 +92,30 @@ def translate_celery_state_naming(job_state: str) -> str:
     return job_state
 
 
-def handle_shieldhit_cancellation(tasks: list[CeleryTaskModel]):
-    """Function triggering shieldhit processes to terminate and dump results"""
-    celery_ids = []
+def handle_cancellation_with_fetching(tasks: list[CeleryTaskModel]):
+    """Function cancel tasks with feching data"""
+    celery_ids_to_terminate = []
+    celery_ids_to_dump_data = []
     for task in tasks:
         if task.task_state == EntityState.RUNNING.value:
-            logging.warning(task.sim_pid)
-            os.kill(task.sim_pid, signal.SIGINT)
-        elif task.task_state in (EntityState.PENDING.value, EntityState.UNKNOWN.value) or task.sim_pid is None:
-            celery_ids.append(task.celery_id)
+            celery_ids_to_dump_data.append(task.celery_id)
+        elif task.task_state in (EntityState.PENDING.value, EntityState.UNKNOWN.value):
+            celery_ids_to_terminate.append(task.celery_id)
+            update_task_state(task=task, update_dict={"task_state": EntityState.CANCELED.value})
 
-    # tell celery to stop all other tasks related to this simulation, being in pending or unknown status
-    celery_app.control.revoke(celery_ids, terminate=True, signal="SIGINT")
+    # terminate tasks which do not start
+    celery_app.control.revoke(celery_ids_to_terminate, terminate=True, signal="SIGINT")
 
-
-def handle_fluka_cancellation(tasks: list[CeleryTaskModel]):
-    """Function cancelling fluka processes with results dump"""
-    FILE_NAME = 'rfluka.stop'
-    celery_ids = []
-    for task in tasks:
-        if task.task_state == EntityState.RUNNING.value:
-            try:
-                if (target_folder := next(Path(task.path_to_sim).glob("fluka_*/"), None)):
-                    (target_folder / FILE_NAME).write_text("")
-            except OSError as e:
-                logging.warning("Error due to file operation: %s", str(e))
-        elif task.task_state in (EntityState.PENDING.value, EntityState.UNKNOWN.value) or task.path_to_sim is None:
-            celery_ids.append(task.celery_id)
-
-    # tell celery to stop all other tasks related to this simulation, being in pending or unknown status
-    celery_app.control.revoke(celery_ids, terminate=True, signal="SIGINT")
+    # change celery tasks state to brake a while loop and dump data
+    if len(celery_ids_to_dump_data) > 0:
+        for id in celery_ids_to_dump_data:
+            result = AsyncResult(id)
+            result.backend.store_result(id, None, 'DUMP')
```

Comment:
Instead of custom celery state I would prefer to use metadata approach, like described here:

```
@app.task(bind=True)
def long_running_task(self):
    print("Starting task...")
    for i in range(10):
        time.sleep(1)
        print(f"Processing step {i+1}/10...")

        # Check task metadata for a "dump" command
        task_info = self.backend.get_task_meta(self.request.id)
        if task_info.get("dump"):
            print("Dump signal received. Dumping data...")
            # Perform dump logic here
            self.backend.store_result(self.request.id, {"dump": False}, state="RUNNING")

    print("Task completed.")

# Sending a "dump" signal
def send_dump_signal(task_id):
    result = AsyncResult(task_id)
    result.backend.store_result(task_id, {"dump": True}, state="RUNNING")

# Example usage
task = long_running_task.apply_async()
send_dump_signal(task.id)
```
It's quite dangerous to introduce own states...

Reply:
Check which backends support metadata, redis and DB should be OK, rabbitmq probably not...

Reply:
Why do you think a custom states could be dangerous?

Reply:
I think we can stick with something based on `self.backend.store_result` now and think of more generic solution in the next step.

Reply:
> Why do you think a custom states could be dangerous?

For couple of reasons:
1. Consistency: we already have simulation and backend states (and their transitions) defined in the backend model. Introducing more states which are inconsistent from what we have a model is an issue.
2. State transitions in celery. Celery itself knows about transitions between built-in states and uses that information in many places. See https://docs.celeryq.dev/en/stable/userguide/tasks.html#task-states and https://docs.celeryq.dev/en/stable/reference/celery.states.html. Introduction a state with unknow transition may lead to undefined behaviour.
3. Complexity. Introducing new state increases complexity of the state model.